src/components/explorers/UCB_explorer.jl

```diff
@@ -37,7 +39,9 @@ See more details at Section (2.7) on Page 35 of the book *Sutton, Richard S., an
 """ function (p::UCBExplorer)(values::AbstractArray)
     v, inds = find_all_max(@. values + p.c * sqrt(log(p.step + 1) / p.actioncounts))
     action = sample(p.rng, inds)
-    p.actioncounts[action] += 1
-    p.step += 1
+    if p.is_training
+        p.actioncounts[action] += 1
+        p.step += 1
+    end
     action
 end
```

Comment:
`testmode!(::UCBExplorer, mode=true)` missing

better to add an error message if it is not extended